src/ddmd/astbase.d

```diff
@@ -5888,6 +5920,31 @@ struct ASTBase
         }
     }
 
+    extern (C++) final class StaticForeach : RootObject
+    {
+        Loc loc;
+
+        ForeachStatement aggrfe;
+        ForeachRangeStatement rangefe;
+
+        final extern (D) this(Loc loc,ForeachStatement aggrfe,ForeachRangeStatement rangefe)
```

Comment:
It seems the rest of dmd has slightly different punctuation conventions - e.g. space after comma and space around operators. I'll let @WalterBright rule on that.